enterprise/app/jobs/captain/conversation/response_builder_job.rb

```diff
@@ -17,6 +18,22 @@ def perform(conversation, assistant)
 
   delegate :account, :inbox, to: :@conversation
 
+  def detect_conversation_language
+    # Skip if language is already detected
+    # If the account has google translate API enabled, it's possible that the language will already be detected
+    return if @conversation.additional_attributes&.dig('conversation_language').present?
+
+    last_message = @conversation.messages.incoming.last
+    return if last_message.blank? || last_message.content.blank?
+
+    language = Captain::Llm::LanguageDetectionService.new.detect(last_message.content)
+    return if language.blank?
+
```

Comment:
I would suggest to send the summary of the conversation. Use conversation.to_llm_text rather than last message.